From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Fri, 5 Jun 2026 17:15:22 +0800
Subject: local-asserth-decls

# DP: Description: /usr/include/assert.h
# DP:	One must be allowed to include <assert.h> multiple times with different
# DP:	values for NDEBUG, so the file is not protected against multiple
# DP:	inclusions.  Unfortunately this means that the declarations for
# DP:	__assert_fail() and the like may occur multiple times in a compilation
# DP:	unit, causing gcc to issue a batch of warnings.
# DP:	I believe this can be fixed by protecting the declarations (but only
# DP:	those declarations) against repetition.
# DP: Author: Jeroen T. Vermeulen <jtv@xs4all.nl>
# DP: Upstream status: Not submitted
# DP: Status Details: Plan to submit
# DP: Date: 2003-01-01

# DP: Description: /usr/include/assert.h
# DP:	One must be allowed to include <assert.h> multiple times with different
# DP:	values for NDEBUG, so the file is not protected against multiple
# DP:	inclusions.  Unfortunately this means that the declarations for
# DP:	__assert_fail() and the like may occur multiple times in a compilation
# DP:	unit, causing gcc to issue a batch of warnings.
# DP:	I believe this can be fixed by protecting the declarations (but only
# DP:	those declarations) against repetition.
# DP: Author: Jeroen T. Vermeulen <jtv@xs4all.nl>
# DP: Upstream status: Not submitted
# DP: Status Details: Plan to submit
# DP: Date: 2003-01-01
---
 assert/assert.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/assert/assert.h b/assert/assert.h
index 53c9aa9..4eb2c2d 100644
--- a/assert/assert.h
+++ b/assert/assert.h
@@ -89,6 +89,8 @@
 
 #else /* Not NDEBUG.  */
 
+#ifndef _ASSERT_H_DECLS
+#define _ASSERT_H_DECLS
 __BEGIN_DECLS
 
 /* This prints an "Assertion failed" message and aborts.  */
@@ -116,6 +118,7 @@ extern _Bool __assert_single_arg (_Bool);
 # endif
 
 __END_DECLS
+#endif /* Not _ASSERT_H_DECLS */
 
 /* When possible, define assert so that it does not add extra
    parentheses around EXPR.  Otherwise, those added parentheses would
